MyOnboarding/src/MyOnboardingApp.Database/Repository/TodoListRepository.cs

```diff
@@ -1,49 +1,69 @@
-            new TodoListItem
-            {
-                Text = "1st Todo Item",
-                Id = new Guid("11111111-1111-1111-1111-aabbccddeeff")
-            },
-            new TodoListItem
-            {
-                Text = "2nd Todo Item",
-                Id = new Guid("22222222-2222-2222-2222-aabbccddeeff")
-            },
-            new TodoListItem
-            {
-                Text = "3rd Todo Item",
-                Id = new Guid("33333333-3333-3333-3333-aabbccddeeff")
-            }
-        };
-
-
-        public async Task<IEnumerable<TodoListItem>> GetAllItemsAsync() 
-            => await Task.FromResult(s_items);
-        
+        private readonly IMongoDatabase _database;
+        private readonly IMongoCollection<TodoListItem> _databaseItems;
+        private const string CollectionName = "TodoItems";
+
+
+        public TodoListRepository(IDatabaseConnection connectionString)
+        {
+            _database = GetDatabaseFromConnection(connectionString);
+            _databaseItems = LoadCollectionFromDatabase(CollectionName);
+        }
+
+
+        public async Task<IEnumerable<TodoListItem>> GetAllItemsAsync()
+        {
+            var results = await _databaseItems.FindAsync(FilterDefinition<TodoListItem>.Empty);
+            return results.ToEnumerable();
+        }
+
 
         public async Task<TodoListItem> GetItemByIdAsync(Guid id) 
-            => await Task.FromResult(s_items[0]);
-        
+            => await _databaseItems.Find(_ => _.Id == id).SingleOrDefaultAsync(CancellationToken.None);
 
-        public async Task<TodoListItem> AddNewItemAsync(TodoListItem newItem) 
-            => await Task.FromResult(s_items[2]);
+
+        public async Task<TodoListItem> AddNewItemAsync(TodoListItem newItem)
+        {
+            await _databaseItems.InsertOneAsync(newItem);
+            return newItem;
+        }
 
 
         public async Task<TodoListItem> ReplaceItemAsync(TodoListItem item)
-            => await Task.FromResult(item);
+        {
+            var filter = Builders<TodoListItem>.Filter.Eq(_ => _.Id, item.Id);
+            var updateDefinition = Builders<TodoListItem>.Update.Set("Text", item.Text).Set("LastUpdateTime", item.LastUpdateTime);
+            var itemToReplace = await _databaseItems.FindOneAndUpdateAsync(filter, updateDefinition);
+            return itemToReplace;
+        }
 
 
         public async Task<TodoListItem> DeleteItemAsync(Guid id) 
-            => await Task.FromResult(s_items[1]);
+            => await _databaseItems.FindOneAndDeleteAsync(_ => _.Id == id);
+
+
+        private static IMongoDatabase GetDatabaseFromConnection(IDatabaseConnection connectionString)
+        {
+            var connection = MongoUrl.Create(connectionString.GetDatabaseConnectionString());
+            var client = new MongoClient(connection);
+            var databaseName = connection.DatabaseName;
+            return client.GetDatabase(databaseName);
```

Comment:
You don't need to store value of a property in a local variable (unless the property is buried too deep in a complex object) - feel free to empty the line above (it is always nice to leave an empty line above a return statement)…
```suggestion
            return client.GetDatabase(connection.DatabaseName);
```